src/blas/backends/rocblas/rocblas_level2.cpp

```diff
@@ -1742,15 +1752,70 @@ TRSV_LAUNCHER_USM(std::complex<double>, rocblas_ztrsv)
 #undef TRSV_LAUNCHER_USM
 
 } // namespace column_major
+
 namespace row_major {
 
 // Buffer APIs
 
+template <typename Func, typename T>
+inline void gemv(Func func, sycl::queue &queue, transpose trans, int64_t m, int64_t n,
+                 std::complex<T> alpha, sycl::buffer<std::complex<T>, 1> &a, int64_t lda,
+                 sycl::buffer<std::complex<T>, 1> &x, int64_t incx, std::complex<T> beta,
+                 sycl::buffer<std::complex<T>, 1> &y, int64_t incy) {
+    auto new_trans = trans == oneapi::mkl::transpose::nontrans ? oneapi::mkl::transpose::trans
+                                                               : oneapi::mkl::transpose::nontrans;
+
+    if (trans == oneapi::mkl::transpose::conjtrans) {
+        alpha = std::conj(alpha);
+        beta = std::conj(beta);
+
+        if (m > 0) {
+            queue.submit([&](sycl::handler &cgh) {
+                const auto abs_incx = std::abs(incx);
+                auto acc = x.template get_access<sycl::access::mode::read_write>(cgh);
+                cgh.parallel_for(sycl::range{ (std::size_t)m }, [=](sycl::id<1> id) {
+                    const auto index = id * abs_incx;
+                    acc[index] = std::conj(acc[index]);
+                });
+            });
```

Comment:
I wonder if it makes sense to define an inline function (or a lambda or maybe even a preprocessor macro) to wrap this, something like:
```c++
template <typename fp>
static inline conj_vector(sycl::queue& queue, sycl::buffer<fp>& x,
                          int64_t m, int64_t incx) {
}
```
then the contents of the function would be lines 1773-1780 above. Then you can reduce a lot of the code duplication below by reusing this function.

Reply:
Done